src/Data/Managers/V2/GetMetadata/useGetMetadata.ts

```diff
@@ -0,0 +1,73 @@
+import { UseQueryResult, useQuery } from "@tanstack/react-query";
+import { ParsedNumber } from "@/Core";
+import { PrimaryBaseUrlManager } from "@/UI";
+import { useFetchHelpers } from "../helpers";
+
+interface Metadata {
+  data: string | undefined;
+}
+/**
+ * Custom hook to fetch metadata for a specific service entity and key.
+ * @param environment - The environment of the service.
+ * @param service_entity - The entity name of the service.
+ * @param service_id - The ID of the service.
+ * @param instanceVersion - The version of the service instance.
+ * @param key - The key for the metadata.
+ * @returns An object containing the custom hook.
+ */
+export const useGetMetadata = (
+  environment: string,
+  service_entity: string,
+  service_id: string,
+  instanceVersion: ParsedNumber | undefined,
```

Comment:
if we not specify the instance version the query will return metadata for the latest version possible. in the current usage we get the version from the fetched data and according to the typing it can be already there undefined, so either way it moving it to the end is fine I guess.

Reply:
Okay, I think making it optional and moving it to the end makes most sense then.